### PR TITLE
Mutation Functions

### PR DESCRIPTION
**Pull request by aoshiro10**
As title.

### CODE REVIEW

File: colors.py [hunk 1]
```diff
@@ -9,11 +10,25 @@ class Colors(Enum):
...
+	Colors.YELLOW,
+	Colors.BLACK,
+	Colors.PURPLE,
+	Colors.WHITE,
+]
+
+def getRandomColor(colorNums):
+	return colors[randint(0, colorNums-1)]
+
+plotColorsDict = {
```
Comment (DavidLBick) [inline colors.py:26]: Nice I forgot you could do that

File: geneticAlgorithm.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 from graphs import *
+from random import random
+import colors
 
 NUM_INDIVIDUALS = int(1e3)
+MUTATION_RATE = 0.05
```
Comment (DavidLBick) [inline geneticAlgorithm.py:6]: Should we make these command line arguments ? We can use argparse package for that

File: main.py [hunk 3]
```diff
@@ -5,12 +5,17 @@
 colorNum = 3
 graphPath = "graph1.gx"
 population = createInitialPopulation(graphPath, colorNum)
-
+bestGraph = None
+generations = 0
 # Careful! This might create infinite graphs
 while True:
+	generations+=1
```
Comment (DavidLBick) [inline main.py:12]: We could set a hyperparameter for maximum iterations and do while True or generations < max_iter

**Review by DavidLBick — APPROVED** [review]

File: geneticAlgorithm.py (hunk 2 above)
Comment (aoshiro10) [reply geneticAlgorithm.py:6]: Yeah, but I think this is fine for now.

File: main.py (hunk 3 above)
Comment (aoshiro10) [reply main.py:12]: Yeah, we probably should do it for graphs that never find a solution. Any ideas on the threshold or how we can calculate it?

**Pull request merged**